lib/kitchen/driver/gce.rb

```diff
@@ -141,26 +141,97 @@ def destroy(state)
         state.delete(:zone)
       end
 
+      def old_disk_configuration_present?
+        !config[:autodelete_disk].nil? || !config[:disk_size].nil? || !config[:disk_type].nil?
+      end
+
+      def new_disk_configuration_present?
+        !config[:disks].nil?
+      end
+
+      def create_disks_config
+        # This can't be present in default_config because we couldn't
+        # determine which disk configuration the user used otherwise
+        disk_default_config = {
+          autodelete_disk: true,
+          disk_size: 10,
+          disk_type: "pd-standard",
+        }
+
+        if old_disk_configuration_present?
+          # If the old disk configuration is used,
+          # we'll convert it to the new one
+          config[:disks] = {}
+          config[:disks][server_name] = {
+            boot: true,
+          }
+          config[:disks][server_name][:autodelete_disk] = config[:autodelete_disk].nil? ? disk_default_config[:autodelete_disk] : config[:autodelete_disk]
+          config[:disks][server_name][:disk_size] = config[:disk_size].nil? ? disk_default_config[:disk_size] : config[:disk_size]
+          config[:disks][server_name][:disk_type] = config[:disk_type].nil? ? disk_default_config[:disk_type] : config[:disk_type]
+          raise "Disk type #{config[:disks][server_name][:disk_type]} is not valid" unless valid_disk_type?(config[:disks][server_name][:disk_type])
+        elsif new_disk_configuration_present?
+          # If the new disk configuration is present, ensure that for
+          # every disk the needed configuration is set
+          boot_disk_counter = 0
+          config[:disks].each do |disk_name, disk_config|
+            # te&/ => te
+            raise "Disk name invalid. Must match #{DISK_NAME_REGEX}." if disk_name.to_s.match(DISK_NAME_REGEX).to_s.length < disk_name.length
```

Comment:
Recommend moving the `if` logic here and the `DISK_NAME_REGEX` to a `valid_disk_name?` method, in the same way the logic for disk types is wrapped in `valid_disk_type?`